eudemo/eudemo_tests/test_maintenance.py

```diff
@@ -52,3 +66,83 @@ def test_dummy_blanket_port_opt(self):
         solution = design_problem.optimise()
 
         assert design_problem.opt.check_constraints(solution)
+
+
+class TestEquatorialPortDesigner:
+    """Tests the Equatorial Port Designer"""
+
+    def setup_method(self) -> None:
+        """Set-up Equatorial Port Designer"""
+        params = {
+            "ep_height": {"value": 0, "unit": "m"},
+        }
+
+        self.designer = EquatorialPortDesigner(params, None, 0, 0, 0)
+
+    @pytest.mark.parametrize(
+        "xi, xo, zh", zip([2.0, 3.0, 1.0], [9.0, 9.0, 4.0], [5.0, 4.0, 2.0])
+    )
+    def test_ep_designer(self, xi, xo, zh):
+        """Test Equatorial Port Designer"""
+        param_values = {"ep_height": zh}
+        self.designer.params.update_values(param_values)
+        self.designer.x_ib = xi
+        self.designer.x_ob = xo
+        output = self.designer.execute()
+        x_len = xo - xi
+        z_len = zh
+
+        assert math.isclose(output.length, 2 * (x_len + z_len))
+        assert math.isclose(output.area, x_len * z_len)
+
+
+class TestCastellationBuilder:
+    """Tests the Castellation Builder"""
+
+    def setup_method(self) -> None:
+        """Set-up Castellation Builder"""
+        params = {
+            "n_components": {"value": 10, "unit": ""},
+            "cst_r_corner": {"value": 0, "unit": "m"},
+        }
+        length = 1
+        y = (0, 1, 1, 0)
+        z = (-1, -1, 1, 1)
+        face = BluemiraFace(make_polygon({"x": 0, "y": y, "z": z}, closed=True))
+        vector = (1, 0, 0)
+        x_offs = [0]
+        c_offs = [0]
+
+        self.builder = CastellationBuilder(
+            params, {}, length, face, vector, x_offs, c_offs
+        )
+
+    @pytest.mark.parametrize(
+        "xi, xo, zh, yw, vec, x_offsets, c_offsets, exp_v",
+        zip(
+            [2.0, 3.0, 1.0],  # x_inboard
+            [9.0, 9.0, 4.0],  # x_outboard
+            [5.0, 4.0, 2.0],  # z_height
+            [3.0, 2.0, 1.0],  # y_widths
+            [(1, 0, 0), (1, 0, 0), (1, 0, 0.5)],  # extrusion vectors
+            [[3.0], [2.0, 4.0], [1.0]],  # x castellation_positions
+            [[1.0], [1.0, 1.0], [0.5]],  # y/z castellation_offsets
+            [185.0, 160.0, 12.521980674],  # volume check value of Eq. Ports
+        ),
+    )
+    def test_cst_builder(self, xi, xo, zh, yw, vec, x_offsets, c_offsets, exp_v):
+        """Test Castellation Builder"""
+        y = (yw / 2, -yw / 2, -yw / 2, yw / 2)
+        z = (-zh / 2, -zh / 2, zh / 2, zh / 2)
+        yz_profile = BluemiraFace(make_polygon({"x": xi, "y": y, "z": z}, closed=True))
+
+        self.builder.length = xo - xi
+        self.builder.face = yz_profile
+        self.builder.vec = vec
+        self.builder.off = x_offsets
+        self.builder.cst = c_offsets
```

Comment:
This makes me uncomfortable :D. Personally, I like to test things via their intended API, here, rejigging stuff on instantiated things is not we normally interact with builders, and it makes knowledge of the internal design of the Builder required to understand the test, as opposed to beneficial.